tuf/api/metadata.py

```diff
@@ -256,6 +256,61 @@ def sign(
 
         return signature
 
+    def verify_delegate(
+        self,
+        role_name: str,
+        delegate: "Metadata",
+        signed_serializer: Optional[SignedSerializer] = None,
+    ):
+        """Verifies that 'delegate' is signed with the required threshold of
+        keys for the delegated role 'role_name'.
+
+        Args:
+            role_name: Name of the delegated role to verify
+            delegate: The Metadata object for the delegated role
+            signed_serializer: Optional; serializer used for delegate
+                serialization. Default is CanonicalJSONSerializer.
+
+        Raises:
+            UnsignedMetadataError: 'delegate' was not signed with required
+                threshold of keys for 'role_name'
+        """
+
+        # Find the keys and role in our metadata
+        role = None
+        if isinstance(self.signed, Root):
+            keys = self.signed.keys
+            role = self.signed.roles.get(role_name)
+        elif isinstance(self.signed, Targets):
+            if self.signed.delegations:
+                keys = self.signed.delegations.keys
+                # Assume role names are unique in delegations.roles: #1426
+                roles = self.signed.delegations.roles
+                role = next((r for r in roles if r.name == role_name), None)
```

Comment:
Also from trishank: "What does this line do? Looks like magic"
My response was:
> it returns the first role in delegations with name role_name (or None if no match).
> 
> I agree it's not pretty and a bit too clever.
> I could do something like this instead:
> 
> ```
> role = None
> for delegated_role in roles:
>     if delegated_role.name == role_name:
>         role = delegated_role
>         break
> ```
> 
> but that's 5 lines of _initialize-then-modify_ antipattern: not the end of the world but not perfect either.
> 
> the real fix is probably #1426 though (make roles a OrderedDict). Then this code becomes readable and efficient:
> role = roles.get(role_name)

I think I'd leave it as is ... but if others think that's too clever I'm happy to change to the more verbose one

Reply:
Perhaps add a one-liner comment for less familiar readers?
> Find the first role in delegations with name role_name (or None if no match)